cpp/include/cuspatial/hausdorff.hpp

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2019-2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <memory>
+#include <cudf/types.h>
+
+namespace cuspatial {
+
+/**
+ * @brief computes Hausdorff distances for all pairs in a collection of spaces
+ *
+ * https://en.wikipedia.org/wiki/Hausdorff_distance
+ *
+ * Example in 1D (this function operates in 2D):
+ * ```
+ * spaces
+ * [0 2 5] [9] [3 7]
+ *
+ * spaces represented as points per space and concatenation of all points
+ * [0 2 5 9 3 7] [3 1 2]
+ *
+ * cartesian product of all
+ * points visually seperated
```

Comment:
Suggest installing a spell checker in your IDE.

```suggestion
 * points visually separated
```

Reply:
But who checks the spellchecker?

I need to check mine. It apparently doesn't look at comments. :(

Reply:
I use "code spell checker" for vscode.